samza-core/src/main/scala/org/apache/samza/config/JobConfig.scala

```diff
@@ -186,4 +189,6 @@ class JobConfig(config: Config) extends ScalaMapConfig(config) with Logging {
   def getNonLoggedStorePath = getOption(JobConfig.JOB_NON_LOGGED_STORE_BASE_DIR)
 
   def getLoggedStorePath = getOption(JobConfig.JOB_LOGGED_STORE_BASE_DIR)
+
+  def getMetadataStoreFactory = getOption(JobConfig.METADATA_STORE_FACTORY).getOrElse(classOf[CoordinatorStreamStoreFactory].getCanonicalName)
```

Comment:
Should not be defaulted here. Let's do it at the call site where CoordinatorStreamStoreFactory is used.

Reply:
1. CoordinatorStreamStore is not necessarily the long term default implementation, and it's not even the only current implementation. Wouldn't you need to call this with ZkStore for all standalone call sites anyway?

2. Use getName() for class names that are meant to be classloaded.